From a684468de2545e58d558558ee613bc5dbcde76c8 Mon Sep 17 00:00:00 2001
From: Mark Stapp <mjs@cisco.com>
Date: Wed, 4 Dec 2024 15:10:24 -0500
Subject: [PATCH] zebra: improve an rnode debug

Improve a debug when we create a new rib_dest by calling the debug
after setting up the dest.

Signed-off-by: Mark Stapp <mjs@cisco.com>
---
 zebra/zebra_rib.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/zebra/zebra_rib.c b/zebra/zebra_rib.c
index 8aeae3258..f151ef214 100644
--- a/zebra/zebra_rib.c
+++ b/zebra/zebra_rib.c
@@ -4010,10 +4010,10 @@ static void rib_link(struct route_node *rn, struct route_entry *re, int process)
 
 	dest = rib_dest_from_rnode(rn);
 	if (!dest) {
+		dest = zebra_rib_create_dest(rn);
+
 		if (IS_ZEBRA_DEBUG_RIB_DETAILED)
 			rnode_debug(rn, re->vrf_id, "rn %p adding dest", rn);
-
-		dest = zebra_rib_create_dest(rn);
 	}
 
 	re_list_add_head(&dest->routes, re);
-- 
2.39.5

